proxy/src/control/discovery.rs

```diff
@@ -188,7 +189,12 @@ where
                 let service = self.bind.bind(&addr).map_err(|_| ())?;
 
                 Ok(Async::Ready(Change::Insert(addr, service)))
-            }
+            },
+            Update::ChangeMetadata(addr) => {
+                let service = self.bind.bind(&addr).map_err(|_| ())?;
```

Comment:
* When would this fail?
* What happens if this fails? Will things continue as normal or will everything screech to a halt?
* If this fails, would it be better to just ignore the metadata update?

I see that the `Update::Insert()` case does something similar so no change is needed here in this PR, but this pattern seems questionable and deserves to at least have its rationale documented in a comment.

Reply:
@briansmith I'm honestly not sure about this --- you're correct that this behaviour was already present in the `Insert` case before this PR. I wish there was a comment here explaining why this is okay.

Reply:
Let's track it in a follow-up issue.